### DISCUSSION TITLE
Action doesn't seem to do the entire yml for Windows, but works on Linux

### DISCUSSION THREAD

**Discussion by djotaku** [community/community#25883] [Actions]
<p>I’ve got: <a href="https://github.com/djotaku/amortization/blob/master/.github/workflows/windowsbuild.yml" rel="nofollow noopener">https://github.com/djotaku/amortization/blob/master/.github/workflows/windowsbuild.yml</a> and <a href="https://github.com/djotaku/amortization/blob/master/.github/workflows/pullrequest.yml" rel="nofollow noopener">https://github.com/djotaku/amortization/blob/master/.github/workflows/pullrequest.yml</a> which are the same except the first is supposed to run on Windows. (And also I made the first currently run on push so that I don’t have to make fake PRs to test it). When do a push, the Windows action seems to stop after Pytest instead of continuing through to the artifact like the Linux version does.</p>
<p>Ideas what I could be doing wrong?</p>
<p>Thanks!</p>
<pre><code> 
</code></pre>

**Comment by ethomson** [comment, marked as answer]
<p>This is often an issue in how CMD scripts interact with executing other CMD scripts.  You can change the shell that the workflow is using, which should help.  For example:</p>
<pre><code>- name: Test with pytest
  run: |
    pip install pytest
    pytest test.py
  shell: bash
</code></pre>

**Comment by djotaku** [comment]
<p>If I understand correctly, you’re saying that if two Github actions are running at once they end up running the same thing unless you specify a shell?</p>

**Comment by ethomson** [comment]
<p>No, that’s not what I’m saying.  Sometimes invoking a CMD script from <em>within</em> a CMD script causes problems.</p>
<p>You can avoid this by using the CALL command, or you can avoid this by using a different shell altogether.</p>
